wilds/datasets/rxrx1_dataset.py

```diff
@@ -0,0 +1,217 @@
+import os
+from pathlib import Path
+from collections import defaultdict
+
+from PIL import Image
+import pandas as pd
+import numpy as np
+import torch
+
+from wilds.datasets.wilds_dataset import WILDSDataset
+from wilds.common.grouper import CombinatorialGrouper
+from wilds.common.metrics.all_metrics import Accuracy
+
+
+class RxRx1Dataset(WILDSDataset):
+    """
+    The RxRx1-WILDS dataset.
+    This is a modified version of the original RxRx1 dataset.
+
+    Supported `split_scheme`:
+        'official' or 'in-dist'
+
+    Input (x):
+        3-channel fluorescent microscopy images of cells
+
+    Label (y):
+        y is one of 1,139 classes:
+        - 0 to 1107: treatment siRNAs
+        - 1108 to 1137: positive control siRNAs
+        - 1138: negative control siRNA
+
+    Metadata:
+        Each image is annotated with its experiment, plate, well, and site, as
+        well as with the id of the siRNA the cells were perturbed with.
+
+    Website:
+        https://www.rxrx.ai/rxrx1
+        https://www.kaggle.com/c/recursion-cellular-image-classification
+
+    FIXME
+    Original publication:
+        @article{,
+            title={},
+            author={},
+            journal={},
+            year={}
+        }
+
+    License:
+        This work is licensed under a Creative Commons
+        Attribution-NonCommercial-ShareAlike 4.0 International License. To view
+        a copy of this license, visit
+        http://creativecommons.org/licenses/by-nc-sa/4.0/.
+    """
+    _dataset_name = 'rxrx1'
+    _versions_dict = {
+        '1.0': {
+            'download_url': 'https://worksheets.codalab.org/rest/bundles/0x6b7a05a3056a434498f0bb1252eb8440/contents/blob/',
+            'compressed_size': 7_413_123_845}
+    }
+
+    def __init__(self, version=None, root_dir='data', download=False,
+                 split_scheme='official'):
+
+        self._version = version
+        self._split_scheme = split_scheme
+        if self._split_scheme not in ['official', 'in-dist']:
+            raise ValueError(f'Split scheme {self._split_scheme} not recognized')
+
+        # path
+        self._data_dir = Path(self.initialize_data_dir(root_dir, download))
+
+        # Load splits
+        df = pd.read_csv(self._data_dir / 'metadata.csv')
+
+        # Splits
+        if split_scheme == 'official':
+            # Training:   33 experiments, 1 site per experiment (site 1)

```

Comment:
Hmm... reminder me why we want the id_test group in the official split? I don't have the eval numbers when training on just one site, but I imagine they drop.